src/components/ListView.jsx

```diff
@@ -50,21 +48,41 @@ function ListView() {
   //
   const handleCheckboxChange = async (e) => {
     const itemId = e.target.name;
-    //create time variable to save time when user checked the box as purchasedTime
-    const datePurchased = Date.now();
-    //if user want to uncheck the item it can be done and purchasedDate is set to null again
+    // grabbing the specific item from state that is clicked because we will be updating its properties
+    let itemToUpdate = items.find((item) => itemId === item.id);
+
+    const dateOfLastTransaction =
+      itemToUpdate.totalPurchases > 0
+        ? itemToUpdate.purhcasedDate
+        : itemToUpdate.createdAt;
+    const daysSinceLastTransaction =
+      (currentTime - dateOfLastTransaction) / oneDay;
+
+    // if user checks a box, itemToUpdate is taken through this flow
     if (e.target.checked) {
-      setUpdateToDb(localToken, itemId, 'purchasedDate', datePurchased);
+      itemToUpdate = {
+        ...itemToUpdate,
+        newEstimate: calculateEstimate(
```

Comment:
I tried to read through everything quickly, but I may have missed this. What is the purpose of `newEstimate` instead of updating `previousEstimate`? Since `previousEstimate` is what's being passed to `calculateEstimate`, it will always pass in the original `previousEstimate` (frequency) instead of continuing to "learn" as the estimate changes.